test/unit/core.js

```diff
@@ -286,13 +286,19 @@ QUnit.test( "type for `Symbol`", function( assert ) {
 });
 
 QUnit.asyncTest( "isPlainObject", function( assert ) {
-	assert.expect( 15 );
+	assert.expect( 17 );
 
-	var pass, iframe, doc,
+	var pass, iframe, doc, parentObj, childObj,
 		fn = function() {};
 
 	// The use case that we want to match
 	assert.ok( jQuery.isPlainObject( {} ), "{}" );
+	assert.ok( jQuery.isPlainObject( new Object() ), "new Object" ); // jshint ignore:line
+
+	parentObj = { foo: "bar" };
+	childObj = Object.assign( Object.create( parentObj ), { bar: "foo" } );
```

Comment:
Two things: if now it is misidentified it doesn't mean it will be identified with `Object.assign` is used and that assert doesn't have to be proof of the fix, but proof of absence of potential regression.

Those are weak arguments, yeah, but i don't get why it is problematic to check with and without `Object.assign` just in case